src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -2426,6 +2426,30 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_Sve_GetActiveElementCount:
+        {
+            assert(sig->numArgs == 2);
+
+            CORINFO_ARG_LIST_HANDLE arg1     = sig->args;
+            CORINFO_ARG_LIST_HANDLE arg2     = info.compCompHnd->getArgNext(arg1);
+            var_types               argType  = TYP_UNKNOWN;
+            CORINFO_CLASS_HANDLE    argClass = NO_CLASS_HANDLE;
+
+            argType = JITtype2varType(strip(info.compCompHnd->getArgType(sig, arg2, &argClass)));
+            op2     = getArgForHWIntrinsic(argType, argClass);
+            argType = JITtype2varType(strip(info.compCompHnd->getArgType(sig, arg1, &argClass)));
+            op1     = impPopStack().val;
+
+            // Op1 and Op2 are masks. Op1 mask handling will be handled via IsExplicitMaskedOperation.
```

Comment:
What cannot we handle `op2` handling next to the place where we handle `op1`?

Reply:
that would require an `if (id ==NI_Sve_GetActiveElementCount)` in hwintrinsic.cpp. I'm trying to avoid adding any Arm specific code in that file.

Reply:
currently we have this:

```c++
    if (HWIntrinsicInfo::IsExplicitMaskedOperation(intrinsic))
    {
        assert(numArgs > 0);
        GenTree* op1 = retNode->AsHWIntrinsic()->Op(1);
        if (intrinsic == NI_Sve_ConditionalSelect)
        {
            if (op1->IsVectorAllBitsSet() || op1->IsMaskAllBitsSet())
            {
                return retNode->AsHWIntrinsic()->Op(2);
            }
            else if (op1->IsVectorZero())
            {
                return retNode->AsHWIntrinsic()->Op(3);
            }
        }

        if (!varTypeIsMask(op1))
        {
            // Op1 input is a vector. HWInstrinsic requires a mask.
            retNode->AsHWIntrinsic()->Op(1) = gtNewSimdConvertVectorToMaskNode(retType, op1, simdBaseJitType, simdSize);
        }
}

we can have `varTypeIsMask(op2)` after the `op1`, right?

Reply:
Yes, but would still require a `NI_Sve_GetActiveElementCount` check, as we don't want to convert for other nodes.

Reply:
hhm, are there many intrinsic that will have `op2` as mask? we can add assert or something to make sure?

Reply:
I've simplified this handling. There's now just a small extra bit in `IsExplicitMaskedOperation()`